### ISSUE TITLE
[BUG] The bulk() function's return value is always empty

### ISSUE THREAD

**Issue by calebplum** [opensearch-project/opensearch-py#380]
### What is the bug?
When inspecting the return value of the opensearch-py `bulk()` function, it returns the following value (for example when writing 70 items):

```
>>> response = bulk(client, items, index=opensearch_index)
>>> print(response)
(70, [])
```

No matter what I try, the list in the return value's second item is always empty. I'd like to understand what this list is for and how I can handle errors returned from Opensearch.

Is there any existing documentation that describes this? The bulk() function's [documentation](https://opensearch-project.github.io/opensearch-py/api-ref/clients/opensearch_client.html#opensearchpy.OpenSearch.bulk) doesn't have any information about its return value.

### How can one reproduce the bug?
Make a call to the opensearch-py bulk() function and inspect its response.

### What is the expected behavior?
I would like to know what the list in the return value is supposed to represent and how I can check for and handle errors returned from Opensearch.

Also, is there any way to access the underlying response from Opensearch?

### What is your host/environment?
Ubuntu 22.04

### Do you have any screenshots?
N/A

### Do you have any additional context?
N/A

**Comment by saimedhi** [comment]
Hello, @calebplum. Thank you for pointing out the gaps in the documentation. We will update it soon.

I tried using the example in the [userguide](https://github.com/saimedhi/opensearch-py/blob/main/USER_GUIDE.md#adding-documents-in-bulk). I got response as below:

`Adding bulk documents:`
`{'took': 195, 'errors': False, 'items': [{'index': {'_index': 'index-2022-06-08', '_id': '1', '_version': 1, 'result': 'created', '_shards': {'total': 2, 'successful': 1, 'failed': 0}, '_seq_no': 0, '_primary_term': 1, 'status': 201}}, {'index': {'_index': 'index-2022-06-09', '_id': '2', '_version': 1, 'result': 'created', '_shards': {'total': 2, 'successful': 1, 'failed': 0}, '_seq_no': 0, '_primary_term': 1, 'status': 201}}, {'index': {'_index': 'index-2022-06-10', '_id': '3', '_version': 1, 'result': 'created', '_shards': {'total': 2, 'successful': 1, 'failed': 0}, '_seq_no': 0, '_primary_term': 1, 'status': 201}}]`

Here "took" refers to time taken to process bulk request in milliseconds, 
"errors"= False implies no errors, "errors"=True implies has errors
"items" gives the imformation about each individual operation status.

**Comment by calebplum** [comment]
Thank you for providing that information @saimedhi.

Can you confirm which version of opensearchpy was used in your example? I'm using v2.2.0.

If it's not a version issue, can you please help me troubleshoot why my opensearchpy client might be returning different output to yours?

**Comment by dblock** [comment]
@calebplum is there something we should do to help the next person not run into this issue?

**Comment by calebplum** [comment]
@dblock Here's a couple things I can think of that would help:

- Add the return value to opensearchpy.client.bulk()'s [documentation](https://opensearch-project.github.io/opensearch-py/api-ref/clients/opensearch_client.html#opensearchpy.OpenSearch.bulk)
- Add documentation for opensearch.helpers.bulk(), including its return value. For reference, [here's](https://elasticsearch-py.readthedocs.io/en/v8.7.1/helpers.html#elasticsearch.helpers.bulk) the documentation for the equivalent Elasticsearch method. This doesn't explicitly show its return value, however it does explain what it's supposed to return and how to access errors returned in the Elasticsearch response. It would be helpful to include all these details in the equivalent Opensearch documentation, as well as an error-handling example.
- Include return values for the existing examples in the Opensearch User Guide for both [opensearchpy.client.bulk()](https://github.com/opensearch-project/opensearch-py/blob/main/USER_GUIDE.md#adding-documents-in-bulk) and [opensearchpy.helpers.bulk()](https://github.com/opensearch-project/opensearch-py/blob/main/USER_GUIDE.md#adding-documents-in-bulk-using-helper-functions). It'd also be helpful to include error-handling examples, but I'm not sure if this would be the best place for that. I could probably raise a PR to include the return values if you'd like, but I'm not sure what the error-handling best practices are for these methods.

**Comment by dblock** [comment]
@calebplum Makes sense. Would you care to contribute? I'm going to reopen this issue and rename it to a doc one.

**Comment by saimedhi** [comment]
@calebplum, Please feel free to take up this issue and raise a PR with updated docs or user guide. Thanks :)

**Comment by tannerjones4075** [comment]
I would like to work on this issue. I will work through the documentation.

**Comment by tannerjones4075** [comment]
Could you assign this issue to me?

**Comment by dblock** [comment]
@tannerjones4075 Are you still working on this?

**Comment by tannerjones4075** [comment]
> @tannerjones4075 Are you still working on this?

Yes, I am still working on this.

**Comment by Plutone11011** [comment]
@dblock related to this, I think other methods of opensearch.client may benefit from similarly upgraded documentation, like `delete` or `create` . I would gladly contribute to this, should I open a new issue or a PR referencing this issue is sufficient?

**Comment by wbeckler** [comment]
Feel free to open a related issue. I'm sure your contributions will be
appreciated

**Comment by dblock** [comment]
> @dblock related to this, I think other methods of opensearch.client may benefit from similarly upgraded documentation, like `delete` or `create` . I would gladly contribute to this, should I open a new issue or a PR referencing this issue is sufficient?

Yes please!
